packages/neo4j/src/index.ts

```diff
@@ -0,0 +1,258 @@
+import neo4j from "neo4j-driver"
+import type { Adapter } from "next-auth/adapters"
+import { v4 as uuid } from "uuid"
+
+import { neo4jWrap } from "./utils"
+
+export function Neo4jAdapter(neo4jSession: typeof neo4j.Session): Adapter {
+  return {
+    // * * * * * * * * * * * * * * * *
+    // User
+    // * * * * * * * * * * * * * * * *
+
+    async createUser({ emailVerified, ...userData }) {
+      return await neo4jWrap(
+        neo4jSession,
+        ` CREATE (u:User)
+          SET
+            u += $userData
+            ${
+              undefined !== emailVerified
+                ? `, u.emailVerified = datetime($emailVerified)`
+                : ``
+            }
+          RETURN properties(u) AS u`,
+        {
+          userData: { id: uuid(), ...userData },
+          emailVerified,
+        }
+      )
```

Comment:
```suggestion
export function Neo4jAdapter(neo4jSession: typeof neo4j.Session): Adapter {
  const query = neo4jWrap(neo4jSession)
  return {

    async createUser({ emailVerified, ...userData }) {
      return await query(
        ` CREATE (u:User)
          SET
            u += $userData
            ${
              undefined !== emailVerified
                ? `, u.emailVerified = datetime($emailVerified)`
                : ``
            }
          RETURN properties(u) AS u`,
        {
          userData: { id: uuid(), ...userData },
          emailVerified,
        }
      )
```

Reply:
Good suggestion. I've done this. But, I've run out of time so some of the typescript types are set to any, where I guess they could be more specific.